File: UnivrCore/Sources/UnivrCore/CourseSelectorManager.swift
//
//  CourseSelectorManager.swift
//  UnivrCore
//
//  Created by Leonardo Rossi on 04/03/2026.
//  Copyright (C) 2026 Leonardo Rossi
//  SPDX-License-Identifier: GPL-3.0-or-later
//

import Foundation
import Observation

@MainActor
@Observable
public final class CourseSearchManager {
    public var searchText: String = ""
    public var courses: [Corso] = []
    
    public init() {}
    
    public var filteredCourses: [Corso] {
        Corso.filter(courses, with: searchText)
    }
    
    public func labelForCourse(_ value: String) -> String {
        Corso.label(for: value, in: courses)
    }
    
    public func clearSearch() {
        searchText = ""
    }
}

File: Android/core/src/main/java/dev/leonardorossi/univrcore/CourseSelectorManager.kt
//
//  CourseSelectorManager.kt
//  UnivrCore
//
//  Translated from CourseSelectorManager.swift
//  Copyright (C) 2026 Leonardo Rossi
//  SPDX-License-Identifier: GPL-3.0-or-later
//

package dev.leonardorossi.univrcore

import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.flow.asStateFlow

class CourseSelectorManager {
    private val _searchText = MutableStateFlow("")
    val searchText: StateFlow<String> = _searchText.asStateFlow()

    private val _courses = MutableStateFlow<List<Corso>>(emptyList())
    val courses: StateFlow<List<Corso>> = _courses.asStateFlow()

    val filteredCourses: List<Corso>
        get() = Corso.filter(_courses.value, _searchText.value)

    fun setSearchText(text: String) {
        _searchText.value = text
    }

    fun setCourses(courses: List<Corso>) {
        _courses.value = courses
    }

    fun labelForCourse(value: String): String {
        return Corso.label(value, _courses.value)
    }

    fun clearSearch() {
        _searchText.value = ""
    }
}